pkg/network/opencontrail/virtual_router.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2016 Juniper Networks, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opencontrail
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+
+	"github.com/Juniper/contrail-go-api"
+	"github.com/Juniper/contrail-go-api/types"
+)
+
+type virtualRouterMap map[string]*types.VirtualRouter
+
+type VirtualRouterManager struct {
+	client contrail.ApiClient
+	config *Config
+	vrMap  virtualRouterMap
+}
+
+// Setup
+func NewVirtualRouterManager(client contrail.ApiClient, config *Config) (*VirtualRouterManager, error) {
+	vrMgr := new(VirtualRouterManager)
+	vrMgr.vrMap = make(virtualRouterMap, 0)
+	vrMgr.client = client
+	vrMgr.config = config
+
+	objects, err := client.ListDetail("virtual-router", nil)
+	if err != nil {
+		glog.Error(err)
+		return nil, err
+	}
+
+	for _, object := range objects {
+		vr := object.(*types.VirtualRouter)
+		vrMgr.vrMap[vr.GetVirtualRouterIpAddress()] = vr
+	}
+	return vrMgr, nil
+}
+
+// Add Pod(VM) reference to VirtualRouter in Contrail Config
+func (vrMgr *VirtualRouterManager) addPodRefToVirtualRouter(
+	pod *api.Pod, instance *types.VirtualMachine) bool {
+
+	if pod.Status.HostIP == "" {
+		glog.Infof("HostIP is empty during pod(%s) add/update", pod.Name)
+		return true
```

Comment:
Is this a normal case or an error? If the return is true, the called will display "pod has been added to vrouter".
Can this func return (bool, error)?